index.d.ts

```diff
@@ -0,0 +1,47 @@
+declare module 'svelte-routing' {
+  import { SvelteComponent, SvelteComponentTyped } from 'svelte'
+  
+  interface RouterProps {
+    basepath?: string
+    url?: string
+  }
+
+  interface RouteProps {
+    path?: string
+    component?: SvelteComponent
+  }
+
+  interface RouteSlots {
+    default: {
+      location: RouteLocation
+      params: RouteParams
+    }
+  }
+
+  interface RouteLocation {
+    pathname: string
+    search: string
+    hash?: string
+    state: {
+      [k in string | number]: unknown
+    }
+  }
+
+  interface RouteParams {
+    [param: string]: string;
+  }
+
+  interface LinkProps {
+    to: string
+    replace?: boolean
+    state?: {}
+    getProps?: () => ({})
+  }
+
+	export const Router: SvelteComponentTyped<RouterProps>
+	export const Route: SvelteComponentTyped<RouteProps, Record<string, any>, RouteSlots>
+	export const Link: SvelteComponentTyped<LinkProps>
```

Comment:
Since svelte-routing v1.5 all additional props passed to `Links` are spread on the underlying `<a />`. Right now, any additional props, like `class` would result in a type error. The types for the html tags svelte uses come from the `svelte2tsx` package. To include these types we could do:

```ts
import "svelte2tsx/svelte-jsx";

// ...

export const Link: SvelteComponentTyped<
  Omit<
    LinkProps &
      // Include all <a /> props
      svelte.JSX.HTMLProps<HTMLAnchorElement> &
      svelte.JSX.SapperAnchorProps,
    // Forbid overriding the href prop, since it is generated by svelte-routing
    "href"
  >
>
```